Assets/Scripts/Ship/ShipVisualHandler.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+[Serializable]
+public enum ShipModuleType {
+    Canon,
+    Mast
+    //TODO: Add all modules types
+}
+
+[Serializable]
+public struct ShipModule {
+    public GameObject prefab;
+    public double cost;
+    public ShipModuleType type;
```

Comment:
Trying to think through how we'll use this, so for example, to enable a player to add Cannon's to their ship, it'll look like:

1. Define a module with the cannon model, whatever we want the cost to be, and the `ShipModuleType.Canon` type
2. When player "purchases" this module, we'll append it to the `modules` list for the player's ship
3. <TBD> We'll add a new firing location (or enable some existing location) in the PlayerFiring script that corresponds to this cannon module

Am I getting that right?

Reply:
Yes, if you look at `ShipModuleLocation` where the current ship modules are stored you will see that the `ShipModule` can be `null`, if it is than visually the module will be empty but once the player buys something it will show in the relative location or replace the module